packages/synapse-react-client/src/components/Authentication/UsernamePasswordForm.tsx

```diff
@@ -55,20 +54,22 @@ export default function UsernamePasswordForm(props: UsernamePasswordFormProps) {
         value={password}
         onChange={e => setPassword(e.target.value)}
       />
-      <Link href={resetPasswordUrl}>Forgot password?</Link>
+      {!hideForgotPasswordButton && (
+        <Link href={resetPasswordUrl}>Forgot password?</Link>
+      )}
```

Comment:
Allow hiding the forgot password button, which would never be used when resetting 2FA (you must have your 2FA code to change your password)